src/envoy/http/authn/peer_authenticator.cc

```diff
@@ -36,25 +36,25 @@ bool PeerAuthenticator::run(Payload* payload) {
   if (policy_.peers_size() == 0) {
     ENVOY_LOG(debug, "No method defined. Skip source authentication.");
     success = true;
```

Comment:
Remove assignment and return true directly. Move declaration of success after  if.